src/app/obfboard.ts

```diff
@@ -79,18 +128,42 @@ export class Button {
 }
 
 export class Image {
+
+  @OneOf(['url', 'data', 'path'], { message: 'Image with id "$value" must specifiy data, a url or a path' })
+  @IsString({ message: 'Image id must be a string' })
+  @IsNotEmpty({ message: 'Image id must be specified'})
   id: string;
+
+  @IsOptional()
+  @IsInt()
   width: number;
+
+  @IsOptional()
+  @IsInt()
   height: number;
+
+  @IsOptional()
   data: string;
+
+  @IsOptional()
+  @IsUrl()
   url: string;
+
+  @IsOptional()
+  @IsString()
   path: string;
+
   contentType: string;
+
+  @IsDefined()
   parent: OBFBoard;
+
+  @IsOptional()
+  @IsString()
   svgData: string;
 
   deserialize(input: any, parent: OBFBoard): Image {
-    this.id = String(input.id);
+    this.id = input.id && String(input.id);
```

Comment:
This won't work as expected if input.id is 0
```suggestion
     this.id = (input.id || input.id === 0) && String(input.id);
```